katana/native/cases/static/cases/js/casesApp.js

```diff
@@ -0,0 +1,1040 @@
+		"retry": { "@type": "If", "@Condition": "", "@Condvalue": "", "@count": "0", "@interval": "0"}, 
+	 };
+	 return newCaseStep;
+},
+
+
+
+	addStepToCase: function(){
+	// Add an entry to the jsonTestSuites....
+	var newCaseStep = createNewStep();
+	if (!caseApp.jsonCaseSteps['step']) {
+		caseApp.jsonCaseSteps['step'] = [];
+		}
+	if (!jQuery.isArray(caseApp.jsonCaseSteps['step'])) {
+		caseApp.jsonCaseSteps['step'] = [caseApp.jsonCaseSteps['step']];
+		}
+	caseApp.jsonCaseSteps['step'].push(newCaseStep);
+	caseApp.mapCaseJsonToUi(caseApp.jsonCaseSteps);
+},
+
+// Save UI Requirements to JSON table. 
+	saveUItoRequirements: function(){
+	rdata= caseApp.jsonCaseObject['Requirements']['Requirement'];
+	rlen = Object.keys(rdata).length;
+	console.log("Number of Requirements = " + rlen );
+	console.log(rdata);
+	for (var s=0; s<Object.keys(rdata).length; s++ ) {
+				console.log("Requirements before save "+rdata[s]);
+				rdata[s] = katana.$activeTab.find("#textRequirement-"+s+"-id").attr('value');
+				console.log("Requirements after save "+rdata[s]);
+		}
+
+},
+
+	 createRequirementsTable: function(){
+	var items =[]; 
+	katana.$activeTab.find("#tableOfCaseRequirements").html("");  // This is a blank div. 
+	items.push('<table id="Requirements_table_display" class="case_req_configuration_table  striped" width="100%" >');
+	items.push('<thead>');
+	items.push('<tr id="ReqRow"><th>#</th><th>Requirement</th><th/><th/></tr>');
+	items.push('</thead>');
+	items.push('<tbody>');
+	console.log("createRequirementsTable");
+	caseApp.adjustRequirementsTable();
+	rdata = caseApp.jsonCaseObject['Requirements']['Requirement'];
+	console.log(rdata, Object.keys(rdata).length, caseApp.jsonCaseObject, caseApp.jsonCaseObject['Requirements']['Requirement']);
+					
+	for (var s=0; s<Object.keys(rdata).length; s++ ) {
+				var oneReq = rdata[s];
+				console.log("oneReq", oneReq);
+				var idnumber = parseInt(s) + 1; 
+
+				if (oneReq == null) {
+					oneReq = '';
+				}
+
+			items.push('<tr data-sid=""><td>'+idnumber+'</td>');
+			var bid = "textRequirement-name-"+s+"-id";	
+				
+			items.push('<td><input type="text" value="'+oneReq+'" id="'+bid+'"/></td>');
+		
+			bid = "deleteRequirement-"+s+"-id";
+			items.push('<td><i  class="fa fa-trash"  title="Delete" id="'+bid+'" katana-click="caseApp.deleteOneRequirementToLine()" key="'+bid+'"/>');
+			bid = "saveOneRequirement-"+s+"-id";
+			var tbid = "textRequirement-name-"+s+"-id";	
+			items.push('<i class="fa fa-pencil" title="Save Edit" id="'+bid+'" txtId="'+tbid+'" katana-click="caseApp.saveRequirementToLine()" key="'+bid+'"/>');
+			bid = "insertRequirement-"+s+"-id";
+			items.push('<i class="fa fa-plus"  title="Insert" id="'+bid+'" katana-click="caseApp.insertRequirementIntoLine()" key="'+bid+'"/></td>');
+
+		}
+	
+	items.push('</tbody>');
+	items.push('</table>');
+		
+	katana.$activeTab.find("#tableOfCaseRequirements").html(items.join(""));  //
+},
+
+	adjustRequirementsTable: function(){
```

Comment:
Incorporate a callback so that a sequence in which JS lines of code execution is established.

Reply:
The function called is too short to warrant complexity of callbacks in the code.

Reply:
How about chaining?